src/vehicle.cpp

```diff
@@ -5596,15 +5627,195 @@ void vehicle::refresh()
+                break;
+            }
+        }
+        // add fake mounts based on the edge info
+        if( edge_info.is_edge_mount() ) {
+            if( edge_info.forward == -1 && fake_mounts.find( rp.first + point_east ) == fake_mounts.end() ) {
+                fake_vehicle_mount fake_forward( rp.first + point_east, rp.first, false, disp, struc );
+                fake_mounts.emplace( fake_forward.mount, fake_forward );
+            }
+            if( edge_info.back == -1 && fake_mounts.find( rp.first + point_west ) == fake_mounts.end() ) {
+                fake_vehicle_mount fake_back( rp.first + point_west, rp.first, false, disp, struc );
+                fake_mounts.emplace( fake_back.mount, fake_back );
+            }
+            if( edge_info.left == -1 && fake_mounts.find( rp.first + point_north ) == fake_mounts.end() ) {
+                fake_vehicle_mount fake_left( rp.first + point_north, rp.first, false, disp, struc );
+                fake_mounts.emplace( fake_left.mount, fake_left );
+            }
+            if( edge_info.right == -1 && fake_mounts.find( rp.first + point_south ) == fake_mounts.end() ) {
+                fake_vehicle_mount fake_right( rp.first + point_south, rp.first, false, disp, struc );
+                fake_mounts.emplace( fake_right.mount, fake_right );
+            }
+            edges.emplace( rp.first, edge_info );
+        }
+    }
+    // update the precalc info for the added fake mounts
+    precalc_fake_mounts( 0, pivot_rotation[0], pivot_anchor[0] );
+    // install the fake parts into parts and add appropriate entires to relative_parts
+    for( std::pair<const point, fake_vehicle_mount> &fm : fake_mounts ) {
+        vehicle_part vp( parts[fm.second.visible_part] );
+        vp.mount = fm.first;
+        vp.precalc = fm.second.precalc;
+
+        parts.push_back( vp );
+        fm.second.fake_part_index = parts.size() - 1;
+
+        std::vector<int> relative;
+        relative.push_back( fm.second.fake_part_index );
+        relative_parts.emplace( fm.first, relative );
+
+        ++fake_part_count;
+    }
+}
+
+void vehicle::remove_fake_parts()
+{
+    if( fake_part_count > 0 ) {
+        parts.resize( parts.size() - fake_part_count );
+    }
+    for( const std::pair<const point, fake_vehicle_mount> &fm : fake_mounts ) {
+        if( relative_parts.find( fm.first ) != relative_parts.end() ) {
+            relative_parts.erase( fm.first );
+        }
+    }
+    edges.clear();
+    fake_mounts.clear();
+    fake_part_count = 0;
+}
+
+bool vehicle::real_or_active_fake_part( int part_num ) const
+{
+    for( const std::pair<const point, fake_vehicle_mount> &fm : fake_mounts ) {
+        if( fm.second.fake_part_index == part_num ) {
+            return fm.second.active;
+        }
+    }
+    return !fake_part( part_num );
+}
+
+point vehicle::get_abs_diff( const point &one, const point &two ) const
+{
+    return point( std::abs( one.x - two.x ), std::abs( one.y - two.y ) );
```

Comment:
```suggestion
    return ( one - two ).abs();
```
Taken from `manhatten_dist(...)` (defined in "line.cpp").